provider/boltmem/boltmem.go

```diff
@@ -261,15 +261,15 @@ func (s *Silences) Mutes(lset model.LabelSet) bool {
 
 // All returns all existing silences.
 func (s *Silences) All() (*types.SilencesQueryResponse, error) {
-	return s.Query(len(s.k), 0)
+	return s.Query(len(s.sils), 0, types.ByCreatedAt)
```

Comment:
ahhh roger. I need to get that value inside a lock before calling into `Query` or it will deadlock, right? Since the lock is requested also within that function. So ..

```go
s.mtx.RLock()
n := len(s.sils)
s.mtx.RUnlock()
return s.Query(n, 0, types.ByCreatedAt)
```
?